src/cplscheme/CouplingScheme.hpp

```diff
@@ -68,35 +68,29 @@ class CouplingScheme {
       double startTime,
       int    startTimeWindow) = 0;
 
-  /// Returns true, if initialize has been called.
-  virtual bool isInitialized() const = 0;
-
-  /**
-   * @brief Getter for _sendsInitializedData
-   * @returns _sendsInitializedData
-   */
-  virtual bool sendsInitializedData() const = 0;
-
   /**
-   * @brief Getter for _receivesInitializedData
-   * @returns _receivesInitializedData
+   * @brief Receives result of first advance, if this has to happen inside SolverInterface::initialize()
+   *
+   * This is only relevant for the second participant of the SerialCouplingScheme, because other coupling schemes only
+   * receive initial data in initialize. This part is implemented as a public function and has to be called explicitly
+   * from the outer scope (usually SolverInterfaceImpl), because the outer scope has to additional actions between
+   * CouplingScheme::initialize and CouplingScheme::receiveResultOfFirstAdvance
    */
-  virtual bool receivesInitializedData() const = 0;
+  virtual void receiveResultOfFirstAdvance(){
+      //@todo: Add security checks?
+  };
 
   /**
-   * @brief Initializes the data for first implicit coupling scheme iteration.
+   * @brief Performs actions before returning from SolverInterface::initialize()
    *
-   * Has to be called after initialize() and before advance().
-   * If this method is not used, the first participant has zero initial values
-   * for its read data, before receiving data in advance(). If non-zero values
-   * are needed, this has to be configured in the coupling-scheme XML
-   * exchange-data tags. This method can nevertheless also be called if no
-   * initialization is necessary. Then it is simply skipped.
-   * It has to be called after initialize() and before
-   * advance(). The second participant has to write the initial data values
-   * to preCICE after initialize() and before initializeData().
+   * Inside SolverInterface::initialize() one has to differentiate between received initial data and received data from
+   * advance. This difference is unimportant as soon as SolverInterface::initialize() and, therefore, cleaned up
+   * by this function.
    */
-  virtual void initializeData() = 0;
+  virtual void completeInitialization() = 0;
```

Comment:
... as soon as what? Sth is missing here.

Reply:
I now entirely removed this function by merging #1362.